base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -1512,9 +1514,17 @@ fn handle_possible_reorg<T: BlockchainBackend>(
 
     let num_added_blocks = reorg_chain.len();
     let removed_blocks = reorganize_chain(db, block_validator, fork_height, &reorg_chain)?;
-
     let num_removed_blocks = removed_blocks.len();
 
+    if let Some(block) = removed_blocks.first() {
```

Comment:
I think this makes more sense just before the last line of `reorganize_chain`. That method is already managing when the db transactions are written, so it's easier to find, and also easier think of `reorganize_chain` as a single DB operation

Reply:
Sounds good, will take a look